client/my-sites/plugins/main.jsx

```diff
@@ -567,12 +289,15 @@ export default React.createClass( {
 		return some( this.props.sites.getSelectedOrAllWithPlugins(), site => site && site.jetpack && site.canUpdateFiles );
 	},
 
-	updateAllPlugins() {
-		PluginsActions.removePluginsNotices( this.state.notices.completed.concat( this.state.notices.errors ) );
-		this.state.plugins.forEach( plugin => {
-			plugin.sites.forEach( site => PluginsActions.updatePlugin( site, site.plugin ) );
-		} );
-		this.recordEvent( 'Clicked Update all Plugins', true );
+	showPluginListPlaceholders( isWpCom ) {
```

Comment:
I like the idea of passing the required only the required sites and plugins the component should care about. Instead of all the plugins and all sites. 
Ideally we would have some sort of middle state where we can tell is a site has loaded the plugins or not or if the array is just empty because it is supposed to be. 

And since we currently listen for changes to the store here I think it might be a good place to leave it here.